presto-main/src/main/java/com/facebook/presto/cost/CostComparator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.cost;
+
+import com.facebook.presto.spi.statistics.Estimate;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+
+import javax.inject.Inject;
+
+import java.util.Comparator;
+
+public class CostComparator
+        implements Comparator<PlanNodeCostEstimate>
+{
+    private final double cpuWeight;
+    private final double memoryWeight;
+    private final double networkWeight;
+
+    @Inject
+    public CostComparator(FeaturesConfig featuresConfig)
+    {
+        this(featuresConfig.getCpuCostWeight(), featuresConfig.getMemoryCostWeight(), featuresConfig.getNetworkCostWeight());
+    }
+
+    public CostComparator(double cpuWeight, double memoryWeight, double networkWeight)
+    {
+        this.cpuWeight = cpuWeight;
+        this.memoryWeight = memoryWeight;
+        this.networkWeight = networkWeight;
+    }
+
+    @Override
+    public int compare(PlanNodeCostEstimate left, PlanNodeCostEstimate right)
```

Comment:
The cost comparator should also consider whether a plan exceeds any known physical limits for Presto.  e.g. maybe we're not weighing the memory cost very high, but if one plan exceeds query.max-memory-per-node, we still want to say that it's worse.